.github/workflows/rust.yml

```diff
@@ -29,28 +29,35 @@ jobs:
           DO_BENCH: true
         run: ./contrib/test.sh
 
-  wasm_pack:
+  wasm:
     name: Stable - Docs / WebAssembly Build
-    runs-on: ubuntu-latest
     strategy:
       matrix:
-        rust:
-          - stable
+        target: [ x86_64-unknown-linux-gnu, x86_64-apple-darwin ]
+        include:
+          - target: x86_64-unknown-linux-gnu
+            os: ubuntu-latest
+          - target: x86_64-apple-darwin
+            os: macos-latest
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout Crate
         uses: actions/checkout@v2
       - name: Checkout Toolchain
         uses: actions-rs/toolchain@v1
         with:
           profile: minimal
-          toolchain: ${{ matrix.rust }}
+          toolchain: stable
           override: true
+      - name: Set CC and AR vars for MacOS
+        if: contains(matrix.os, 'macos')
+        run: |
+          echo "CC=/usr/local/opt/llvm/bin/clang" >> $GITHUB_ENV
+          echo "AR=/usr/local/opt/llvm/bin/llvm-ar" >> $GITHUB_ENV
       - name: Building docs
         env:
           DO_DOCS: true
         run: ./contrib/test.sh
-      - name: Install wasm-pack
```

Comment:
Why are we deleting this?

It doesn't actually run anything now: https://github.com/ElementsProject/rust-secp256k1-zkp/runs/3075531208#step:6:68

Also, the script seems to be flawed because it doesn't fail the build if it is not succeeding 😅

Looking at the script, I also saw that this is already setting the CC variable so we either want to remove that or what we do in the CI jobs is pointless.

Reply:
I guess the original is broken as well https://github.com/rust-bitcoin/rust-secp256k1/pull/254. We also don't install llvm? did this every work in CI?

Reply:
You still need wasm-pack to run the wasm tests :)

Reply:
added on line 60 of 0b4c3dbb34b2ec8fd118aa2a2ea46d6034e26cee

Reply:
> We also don't install llvm?

I think llvm just comes pre-installed as @bonomat discovered [here](https://github.com/comit-network/waves/pull/170).

Reply:
See the CI run in this repository: https://github.com/ElementsProject/rust-secp256k1-zkp/pull/24/checks?check_run_id=3082712519#step:6:4

Reply:
It seems to be specific to my machine (don't need set -e in the mac os CI):

https://github.com/rishflab/rust-secp256k1-zkp/pull/1/commits/e1e86ffeac0b0324175bfd421e10c4bf1f2d54d6
https://github.com/rishflab/rust-secp256k1-zkp/pull/1/checks?check_run_id=3099601565

https://github.com/rishflab/rust-secp256k1-zkp/pull/1/commits/298c2944d57a41dfeb24e132e88d08aca4f8a49e
https://github.com/rishflab/rust-secp256k1-zkp/pull/1/checks?check_run_id=3099702619